codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/core/KotlinSymbolProvider.kt

```diff
@@ -237,6 +244,37 @@ class KotlinSymbolProvider(private val model: Model, private val settings: Kotli
         }
     }
 
+    private fun getDefaultValueForTimestamp(node: NumberNode): String {
+        val instant = RuntimeTypes.Core.Instant
+
+        return if (node.isFloatingPointNumber) {
+            val fromEpochMilliseconds = RuntimeTypes.Core.fromEpochMilliseconds
+
+            val value = node.value as Double
+            val ms = round(value * 1e3).toLong()
+            "$fromEpochMilliseconds.invoke($instant, $ms)"
+        } else {
+            "$instant.fromEpochSeconds(${node.value}, 0)"
+        }
+    }
+
+    private fun getDefaultValueForDocument(node: Node): String {
+        val documentSymbol = RuntimeTypes.Core.Content.Document.fullName
+        val content: String = when {
+            node.isNumberNode -> node.asNumberNode().get().value.toString()
+            node.isStringNode -> node.asStringNode().get().value.dq()
+            node.isBooleanNode -> node.asBooleanNode().get().value.toString()
+            node.isNullNode -> "null"
+            // Note: Only empty maps and lists can be used as a default value for DocumentShape
+            // https://smithy.io/2.0/spec/type-refinement-traits.html#default-value-constraints
+            node.isArrayNode -> "listOf()"
+            node.isObjectNode -> "mapOf()"
+            else -> throw RuntimeException("Unsupported node $node")
+        }
```

Comment:
**Nit**: We can avoid the `else` branch and get warned of future document types by switching on `node.type` instead of the boolean properties:

```kotlin
val content = when (node.type) {
    NodeType.NUMBER -> ...
    NodeType.STRING -> ...
    NodeType.BOOLEAN -> ...
    ...
}
```